qrexec/tests/policy_parser.py

```diff
@@ -1026,144 +1026,194 @@ def test_037_eval_ask_no_targets(self):
         with self.assertRaises(exc.AccessDenied):
             policy.evaluate(_req('test-vm3', '@default'))
 
+    def test_038_eval_resolve_dispvm_from_any(self):
+        policy = parser.TestPolicy(policy='''\
+            * * @anyvm @dispvm allow''')
+        resolution = policy.evaluate(_req('test-vm3', '@dispvm'))
+
+        self.assertIsInstance(resolution, parser.AllowResolution)
+        self.assertEqual(resolution.rule, policy.rules[0])
+        self.assertEqual(resolution.target, '@dispvm:default-dvm')
+        self.assertEqual(resolution.request.target, '@dispvm')
+
+    @unittest.expectedFailure
+    def test_039_eval_to_dom0(self):
+        policy = parser.TestPolicy(policy='''\
+            * * test-vm3 @adminvm allow''')
+        resolution = policy.evaluate(_req('test-vm3', 'dom0'))
+
+        self.assertIsInstance(resolution, parser.AllowResolution)
+        self.assertEqual(resolution.rule, policy.rules[0])
+        self.assertEqual(resolution.target, '@adminvm')
+        self.assertEqual(resolution.request.target, 'dom0')
```

Comment:
@woju I think we discussed this, but I'd like to confirm: when VM makes a call with `dom0` target, it gets translated to `@adminvm` very early. This includes "intended_target" argument to the actual call (it gets `keyword adminvm` instead of `name dom0` there).

While it many cases it makes the service script simpler (one case instead of two).
I'm slightly worried about future cases, where there may be more than one AdminVM (think Qubes Air). In that case `@adminvm` should mean "local AdminVM", but maybe it should be possible to address other AdminVM too? Should we assume it will be called something else than bare `dom0` and consider then `dom0` an abstract, deprecated name for `@adminvm`?

Reply:
I think that's accurate, especially the part about `dom0` being deprecated alias. IIRC for non-local AdminVMs we thought about something like `@adminvm:<whatever>`, but because of lack of specifics and no current use, there wasn't much thinking anyway. I may have implicitly assumed that this will be a keyword and not a name, but I'm not sure about that.

Reply:
Ok, makes sense. So, I need to adjust qubesd admin api part to understand `@adminvm` target (since client provided `dom0` is translated to `@adminvm`).